tests/integration_tests/api_proxy_test.rs

```diff
@@ -0,0 +1,71 @@
+use crate::helpers::{COUNTER_WAT, TestEnv, install_canister};
+use anyhow::{Context, anyhow, bail};
+use candid::Principal;
+use ic_agent::Agent;
+use reqwest::Client;
+use tokio::runtime::Runtime;
+use tracing::info;
+use url::Url;
+
+// Test scenario:
+// - install a counter canister
+// - create an agent to interact with the canister
+// - make API calls (status, query, call, read_state) to the canister
+
+pub fn proxy_api_calls_test(env: &TestEnv) -> anyhow::Result<()> {
+    info!("install counter canister ...");
+    let canister_id = install_canister(
+        &env.pic,
+        Principal::anonymous(),
+        None,
+        wat::parse_str(COUNTER_WAT).unwrap(),
+    );
+
+    info!("create agent to interact with the canister ...");
+    let url = Url::parse(&format!(
+        "http://{}:{}",
+        env.ic_gateway_domain,
+        env.ic_gateway_addr.port(),
+    ))
+    .context(anyhow!("failed to parse URL"))?;
+
+    let http_client = Client::builder()
+        .resolve(&env.ic_gateway_domain, env.ic_gateway_addr)
+        .build()
+        .context("failed to build http client")?;
+
+    let agent = Agent::builder()
+        .with_url(url)
+        .with_http_client(http_client)
+        .build()?;
+
+    info!("test proxying of various API calls ...");
+    let rt = Runtime::new().context("failed to start tokio runtime")?;
```

Comment:
Why do we need to create a runtime in each test instead of running the whole thing under `#[tokio::test]` (in `all_tests.rs`)?

Reply:
No reason, just copy and paste and no thinking :)